libs/ffi/ffi.c

```diff
@@ -240,6 +306,39 @@ int FFI_mitls_connect(struct _FFI_mitls_callbacks *callbacks, /* in */ mitls_sta
     CAMLreturnT(int,ret);
 }
 
+// Called by the host server app, after a client has connected to a socket and the calling server has accepted the TCP connection.
+int FFI_mitls_accept_connected(struct _FFI_mitls_callbacks *callbacks, /* in */ mitls_state *state, /* out */ char **outmsg, /* out */ char **errmsg)
+{
+    CAMLparam0();
+    CAMLlocal1(result);
+    int ret;
+
+    *outmsg = NULL;
+    *errmsg = NULL;
+
+    caml_acquire_runtime_system();
+    result = caml_callback2_exn(*g_mitls_FFI_AcceptConnected, state->fstar_state, PtrToValue(callbacks));
+    if (Is_exception_result(result)) {
+        // Call caml_format_exception(Extract_exception(result)) to extract the exception text
+        ret = 0;
+    } else {
+        // AcceptConnected returns back (Connection.connection * int)
+        value connection = Field(result,0);
+        ret = Int_val(Field(result,1));
+        if (ret == 0) {
+            caml_modify_generational_global_root(&state->fstar_state, connection);
```

Comment:
AcceptConnected is cloned code from FFI_mitls_connect(), above.  They both return a (Connection.connection * int) that encodes both the miTLS state variable for use later, and an integer error status code.  Why not a connection option?  I recall having quite a bit of difficulty passing an option out through the FFI, and we wanted the rich integer error code no matter what.

Yes, the OCaml GC will collect the object that state->fstar_state used to point to, with no further work from us.